src/fetch/PageAction.js

```diff
@@ -0,0 +1,113 @@
+import { pageAction } from "./prompts.js";
+import { logger } from "../log/logger.js";
+import { TagRemovingMinimizer } from "../min/TagRemovingMinimizer.js";
+import { getAI } from '../ai/index.js';
+import { Timer } from "../log/timer.js";
+
+export const PageAction = class {
+  constructor(namespace, prompt) {
+    this.namespace = namespace;
+    this.prompt = prompt;
+    this.ai = getAI()
+  }
+
+  async learn(examples) {
+
+  }
+
+  async execute(page) {
+    const timer = new Timer();
+    const url = page.url;
+    const doc = await page.content();
+    // const min = new TagRemovingMinimizer(['style', 'script', 'meta', 'link']);
+    // const minDoc = await min.min(doc, { timer });
+    const context = {
+      html: doc,
+      prompt: this.prompt
+    }
+
+    let prompts;
+    try {
+      prompts = await pageAction.renderMulti(context, 'html', this.ai);
+    } catch (e) {
+      logger.error(`${this} Error while rendering prompts: ${e}`);
+      return;
+    }
+
+    const commands = [];
+    logger.debug(`${this} analyze chunks for next action (${prompts.length})`);
+
+    for (const prompt of prompts) {
+      let answer;
+      try {
+        answer = await this.ai.ask(prompt, { format: 'json' });
+      } catch(e) {
+        logger.error(`${this} AI error, ignore: ${e}`);
+        continue
+      }
+
+      logger.debug(`${this} Got an answer: ${JSON.stringify(answer.partial)}`);
+
+      if (answer.partial?.actionCommand && answer.partial?.actionArgument) {
+        commands.push({
```

Comment:
the AI call should go into learn(), and then we should remember how do to the action